.ci/test_r_package.sh

```diff
@@ -167,45 +167,43 @@ elif [[ $R_BUILD_TYPE == "cran" ]]; then
     if [[ "${TASK}" == "r-rchk" ]]; then
         echo "Checking R package with rchk"
         mkdir -p packages
-        cp ${PKG_TARBALL} packages
+        cp "${PKG_TARBALL}" packages
         RCHK_LOG_FILE="rchk-logs.txt"
         docker run \
-            -v $(pwd)/packages:/rchk/packages \
+            -v "$(pwd)"/packages:/rchk/packages \
             kalibera/rchk:latest \
             "/rchk/packages/${PKG_TARBALL}" \
-        2>&1 > ${RCHK_LOG_FILE} \
+         > ${RCHK_LOG_FILE} 2>&1 \
         || (cat ${RCHK_LOG_FILE} && exit 1)
         cat ${RCHK_LOG_FILE}
 
         # the exceptions below are from R itself and not LightGBM:
         # https://github.com/kalibera/rchk/issues/22#issuecomment-656036156
-        exit $(
+        exit "$(
             cat ${RCHK_LOG_FILE} \
             | grep -v "in function strptime_internal" \
             | grep -v "in function RunGenCollect" \
             | grep --count -E '\[PB\]|ERROR'
-        )
+        )"
     fi
 
     # Test CRAN source .tar.gz in a directory that is not this repo or below it.
     # When people install.packages('lightgbm'), they won't have the LightGBM
     # git repo around. This is to protect against the use of relative paths
     # like ../../CMakeLists.txt that would only work if you are in the repo
     R_CMD_CHECK_DIR="${HOME}/tmp-r-cmd-check/"
-    mkdir -p ${R_CMD_CHECK_DIR}
-    mv ${PKG_TARBALL} ${R_CMD_CHECK_DIR}
-    cd ${R_CMD_CHECK_DIR}
+    mkdir -p "${R_CMD_CHECK_DIR}"
+    mv "${PKG_TARBALL}" "${R_CMD_CHECK_DIR}"
+    cd "${R_CMD_CHECK_DIR}"
 fi
 
 # fails tests if either ERRORs or WARNINGs are thrown by
 # R CMD CHECK
-check_succeeded="yes"
-(
-    R CMD check ${PKG_TARBALL} \
-        --as-cran \
-        --run-donttest \
-    || check_succeeded="no"
-) &
+if R CMD check "${PKG_TARBALL}" --as-cran --run-donttest; then
+    export check_r_package="yes"
+else
+    export check_r_package="no"
+fi
```

Comment:
I don't understand this change (and then later putting `check_succeeded` below). What is the purpose of this? Can you please provide the exact text of the error(s) this is attempting to fix?

Reply:
In testing.sh line 207:
    || check_succeeded="no"
       ^-------------^ SC2030 (info): Modification of check_succeeded is local (to subshell caused by (..) group).

Reply:
This change is just not correct. We might be losing the assignment of `check_succeeded` (that looks like a genuine bug), but it's intentional that this is run in a background process (using `&`), for the reasons described in the comment directly below these lines.

It's possible that we could remove this pattern of running `R CMD check` in a background task and refactor this a bit, but if you are not familiar with R or that command then it's going to be difficult for you to do that.

For the purpose of this PR, please just revert these changes related to this `R CMD check` run. I'll try resolving that in a separate PR.

Reply:
check_r_package="yes" && { R CMD check "${PKG_TARBALL}" --as-cran --run-donttest || check_r_package="no"; } &
 can we do this ?

Reply:
No, it's more complex than that and involves changing more than just this line. Please, for the purpose of this PR, just do not touch this part of this script.